apps/admin/pages/api/auth/[...nextauth].ts

```diff
@@ -46,34 +40,67 @@ export const authOptions: AuthOptions = {
     signIn: async ({ user, account, profile }) => {
       //security
       if (user.email === "demo@example.com") {
-        return Promise.resolve(true);
+        return true;
       }
 
       //restrict domain
       if (
         process.env.RESTRICT_EMAIL_DOMAIN &&
         !profile?.email?.endsWith(`@${process.env.RESTRICT_EMAIL_DOMAIN}`)
       ) {
-        return Promise.resolve(false);
+        return false;
       }
 
       //ok
-      return Promise.resolve(true);
+      return true;
     },
-    session: async ({ session, user, token }) => {
-      //always add this, very handy
-      session.userId = user.id;
+    jwt: async ({ token, user, trigger, account }) => {
```

Comment:
this is the key callback for JWT strategy, now the credentials will be probably broken, haven't checked